dkulib/parallelizer/parallelizer.py

```diff
@@ -64,18 +90,18 @@ class DataFrameParallelizer:
         verbose: If True, log raw details on any error encountered along with the error message and error type.
             Else (default) log only the error message and the error type.
             We recommend trying without verbose first. Usually, the error message is enough to diagnose the issue.
-
     """
-
+    # Default number of worker threads to use in parallel - may be tuned by the end user
     DEFAULT_PARALLEL_WORKERS = 4
-    """Default number of worker threads to use in parallel - may be tuned by the end user"""
+    # By default, we assume the function to apply is row-by-row - should be overriden in the batch case
     DEFAULT_BATCH_SUPPORT = False
-    """By default, we assume the function to apply is row-by-row - should be overriden in the batch case"""
-    DEFAULT_BATCH_SIZE = 10
-    """Default number of rows in one batch - may be tuned by the end user"""
-
+    # Default number of rows in one batch - may be tuned by the end user
+    DEFAULT_BATCH_SIZE = 1
```

Comment:
How about keeping `DEFAULT_BATCH_SIZE = 10` but adding a rule in the __init__ so that is  batch_support is False, then batch_size is 1?
IMO that would be more readable. What do you think?

Reply:
Agreed, done